server/src/session_manager.ts

```diff
@@ -98,32 +152,77 @@ export class SessionManager {
         globalThis.window = dom.window as any;
         globalThis.document = dom.window.document;
 
-        const bgConfig = {
-            fetch: (url: any, options: any) => fetch(url, options),
+        let dispatcher: Agent | undefined;
+        if (proxy) {
+            dispatcher = this.getProxyDispatcher(proxy);
+        } else if (process.env.ALL_PROXY) {
```

Comment:
does ALL_PROXY have priority over HTTPS_PROXY/HTTP_PROXY in yt-dlp too?

Reply:
![image](https://github.com/user-attachments/assets/9246ff80-e221-4760-8e01-eff3e96dddd6)

Reply:
not resolved yet: as u can see in the log, the error is `connection to 192.168.1.85 timeout` when **ALL_PROXY=192.168.86:8080** and **HTTP_PROXY=192.168.1.85:8083**, indicating that the HTTP_PROXY is actually preferred. will push a commit to fix it, `process.env.ALL_PROXY` should be a fallback to HTTPS_PROXY & HTTP_PROXY

Reply:
resolved in 74d2ba0.